### PR TITLE
Frontend: com_contact view review

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Review frontend com_contact views

### Testing Instructions

Install sample testing data and navigate all the com_contact frontend views.

### Expected result

All still works

### Actual result

The views have cs errors ;)

### Documentation Changes Required

None

### CODE REVIEW

File: components/com_contact/views/category/tmpl/default_items.php [hunk 1]
```diff
@@ -14,115 +14,113 @@
...
-			</div>
-		<?php endif; ?>
-	</fieldset>
+		<fieldset class="filters btn-toolbar">
+			<?php if ($this->params->get('filter_field')) : ?>
+				<div class="btn-group">
+					<label class="filter-search-lbl element-invisible" for="filter-search">
+						<span class="label label-warning">
+							<?php echo JText::_('JUNPUBLISHED'); ?>
+						</span><?php echo JText::_('COM_CONTACT_FILTER_LABEL') . '&#160;'; ?>
```
Comment (brianteeman) [inline components/com_contact/views/category/tmpl/default_items.php]: shouldnt the php part be on  new line, like line 39 below

File: components/com_contact/views/category/tmpl/default_items.php [hunk 2]
```diff
@@ -14,115 +14,113 @@
...
+				<div class="list-title span<?php echo $contact_width; ?> col-md-<?php echo $contact_width; ?>">
+					<a href="<?php echo JRoute::_(ContactHelperRoute::getContactRoute($item->slug, $item->catid)); ?>">
+						<?php echo $item->name; ?></a>
+					<?php if ($this->items[$i]->published == 0) : ?>
+						<span class="label label-warning"><?php echo JText::_('JUNPUBLISHED'); ?></span>
+					<?php endif; ?>
+					<?php echo $item->event->afterDisplayTitle; ?>
+					<?php echo $item->event->beforeDisplayContent; ?>
+					<?php if ($this->params->get('show_position_headings')) : ?>
+							<?php echo $item->con_position; ?><br />
```
Comment (brianteeman) [inline components/com_contact/views/category/tmpl/default_items.php]: too many tabs

File: components/com_contact/views/category/tmpl/default_items.php [hunk 3]
```diff
@@ -14,115 +14,113 @@
...
+					<?php endif; ?>
+					<?php if ($this->params->get('show_email_headings')) : ?>
+							<?php echo $item->email_to; ?><br />
```
Comment (brianteeman) [inline components/com_contact/views/category/tmpl/default_items.php]: too many tabs

File: components/com_contact/views/category/tmpl/default_items.php [hunk 4]
```diff
@@ -14,115 +14,113 @@
...
+						<?php $location[] = $item->country; ?>
+					<?php endif; ?>
+					<?php echo implode($location, ', '); ?>
+				</div>
+				<div class="span3 col-md-3">
+					<?php if ($this->params->get('show_telephone_headings') && !empty($item->telephone)) : ?>
+						<?php echo JText::sprintf('COM_CONTACT_TELEPHONE_NUMBER', $item->telephone); ?><br />
+					<?php endif; ?>
+					<?php if ($this->params->get('show_mobile_headings') && !empty ($item->mobile)) : ?>
+							<?php echo JText::sprintf('COM_CONTACT_MOBILE_NUMBER', $item->mobile); ?><br />
```
Comment (brianteeman) [inline components/com_contact/views/category/tmpl/default_items.php]: too many tabs

File: components/com_contact/views/featured/tmpl/default_items.php [hunk 5]
```diff
@@ -18,149 +18,130 @@
...
+								<?php echo JText::_('COM_CONTACT_MOBILE'); ?>
+							</th>
+						<?php endif; ?>
+						<?php if ($this->params->get('show_fax_headings')) : ?>
+							<th class="item-phone">
+								<?php echo JText::_('COM_CONTACT_FAX'); ?>
+							</th>
+						<?php endif; ?>
+						<?php if ($this->params->get('show_suburb_headings')) : ?>
+						<th class="item-suburb">
```
Comment (brianteeman) [inline components/com_contact/views/featured/tmpl/default_items.php]: needs extra tab

**Comment by zero-24** [comment]
Thanks @brianteeman i have just fixed the comments.

File: components/com_contact/views/category/tmpl/default_items.php [hunk 6]
```diff
@@ -14,115 +14,114 @@
...
-
-						<?php if ($this->params->get('show_fax_headings') && !empty($item->fax) ) : ?>
-							<?php echo JText::sprintf('COM_CONTACT_FAX_NUMBER', $item->fax); ?><br />
+	<ul class="category row-striped">
+		<?php foreach ($this->items as $i => $item) : ?>
+			<?php if (in_array($item->access, $this->user->getAuthorisedViewLevels())) : ?>
+				<?php if ($this->items[$i]->published == 0) : ?>
+					<li class="row-fluid system-unpublished cat-list-row<?php echo $i % 2; ?>">
+				<?php else : ?>
+					<li class="row-fluid cat-list-row<?php echo $i % 2; ?>" >
```
Comment (Quy) [inline components/com_contact/views/category/tmpl/default_items.php]: Remove space `?>" >`

**Comment by zero-24** [comment]
Fixed thanks @Quy

File: components/com_contact/views/categories/tmpl/default_items.php [hunk 7]
```diff
@@ -11,23 +11,20 @@
...
-			if (!isset($this->items[$this->parent->id][$id + 1]))
-			{
-				$class = ' class="last"';
-			}
-			?>
+		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
+			<?php if (!isset($this->items[$this->parent->id][$id + 1])) : ?>
+				<?php $class = ' class="last"'; ?>
+			<?php endif; ?>
 			<div <?php echo $class; ?> >
```
Comment (Quy) [inline components/com_contact/views/categories/tmpl/default_items.php]: Remove space before and after `<?php echo $class; ?>`.

File: components/com_contact/views/category/tmpl/default_children.php [hunk 8]
```diff
@@ -8,48 +8,45 @@
...
-				<?php endif; ?>
+?>
+<?php $class = ' class="first"'; ?>
+<?php if ($this->maxLevel != 0 && count($this->children[$this->category->id]) > 0) : ?>
+	<ul class="list-striped list-condensed">
+		<?php foreach ($this->children[$this->category->id] as $id => $child) : ?>
+			<?php if ($this->params->get('show_empty_categories') || $child->numitems || count($child->getChildren())) : ?>
+				<?php if (!isset($this->children[$this->category->id][$id + 1])) : ?>
+					<?php $class = ' class="last"'; ?>
+				<?php endif ?>
```
Comment (Quy) [inline components/com_contact/views/category/tmpl/default_children.php]: Add `;`

File: components/com_contact/views/contact/tmpl/default_address.php [hunk 9]
```diff
@@ -65,66 +64,64 @@
...
+	<?php endif; ?>
+	<?php if ($this->contact->webpage && $this->params->get('show_webpage')) : ?>
+		<dt>
+			<span class="<?php echo $this->params->get('marker_class'); ?>">
+			</span>
+		</dt>
+		<dd>
+			<span class="contact-webpage">
+				<a href="<?php echo $this->contact->webpage; ?>" target="_blank" rel="noopener noreferrer" itemprop="url">
+				<?php echo JStringPunycode::urlToUTF8($this->contact->webpage); ?></a>
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default_address.php]: Indent and move `</a>` to the next line.

File: components/com_contact/views/featured/tmpl/default.php [hunk 10]
```diff
@@ -11,28 +11,25 @@
...
-		<?php endif; ?>
-
-		<?php echo $this->pagination->getPagesLinks(); ?>
-	</div>
-<?php endif; ?>
+	<?php if ($this->params->get('show_page_heading') != 0 ) : ?>
+		<h1><?php echo $this->escape($this->params->get('page_heading')); ?></h1>
+	<?php endif; ?>
+	<?php echo $this->loadTemplate('items'); ?>
+	<?php if ($this->params->def('show_pagination', 2) == 1  || ($this->params->get('show_pagination') == 2 && $this->pagination->pagesTotal > 1)) : ?>
```
Comment (Quy) [inline components/com_contact/views/featured/tmpl/default.php]: Remove extra space

File: components/com_contact/views/contact/tmpl/default_profile.php [hunk 11]
```diff
@@ -8,37 +8,34 @@
...
+							<?php $v_http = substr($profile->value, 0, 4); ?>
+							<?php if ($v_http === 'http') : ?>
+								<?php echo '<dd><a href="' . $profile->text . '">' . JStringPunycode::urlToUTF8($profile->text) . '</a></dd>'; ?>
+							<?php else : ?>
+								<?php echo '<dd><a href="http://' . $profile->text . '">' . JStringPunycode::urlToUTF8($profile->text) . '</a></dd>'; ?>
+							<?php endif; ?>
+							<?php break; ?>
+						<?php case 'profile_dob': ?>
+							<?php echo '<dd>' . JHtml::_('date', $profile->text, JText::_('DATE_FORMAT_LC4'), false) . '</dd>'; ?>
+						<?php break; ?>
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default_profile.php]: Indent

**Comment by Heggi93** [comment]
I have not tested this item.<br /><br />No errors found <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17217">issues.joomla.org/tracker/joomla-cms/17217</a>.</sub>

**Comment by Heggi93** [comment]
I have tested this item :white_check_mark: successfully on 48824b11bcb9c5fc454324bdea6038fccd2108e4<br /><br />No errors found <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17217">issues.joomla.org/tracker/joomla-cms/17217</a>.</sub>

File: components/com_contact/views/categories/tmpl/default_items.php [hunk 12]
```diff
@@ -11,23 +11,20 @@
...
-			<div <?php echo $class; ?> >
+		<?php if ($this->params->get('show_empty_categories_cat') || $item->numitems || count($item->getChildren())) : ?>
+			<?php if (!isset($this->items[$this->parent->id][$id + 1])) : ?>
+				<?php $class = ' class="last"'; ?>
+			<?php endif; ?>
+			<div<?php echo $class; ?>>
 			<?php $class = ''; ?>
 				<h3 class="page-header item-title">
 					<a href="<?php echo JRoute::_(ContactHelperRoute::getCategoryRoute($item->id, $item->language)); ?>">
 					<?php echo $this->escape($item->title); ?></a>
```
Comment (Quy) [inline components/com_contact/views/categories/tmpl/default_items.php]: Indent and move `</a>` to next line. Do the same for line 34.

**Comment by ghost** [comment]
@zero-24 can you please Update Instruction which Codestyle-Errors are meant so Tester know what to look for?

File: components/com_contact/views/featured/tmpl/default.php [hunk 13]
```diff
@@ -11,28 +11,27 @@
...
-		<?php if ($this->params->def('show_pagination_results', 1)) : ?>
-			<p class="counter">
-				<?php echo $this->pagination->getPagesCounter(); ?>
-			</p>
-		<?php endif; ?>
-
-		<?php echo $this->pagination->getPagesLinks(); ?>
-	</div>
-<?php endif; ?>
+	<?php if ($this->params->get('show_page_heading') != 0 ) : ?>
```
Comment (Quy) [inline components/com_contact/views/featured/tmpl/default.php]: Remove space after `0`

File: components/com_contact/views/featured/tmpl/default_items.php [hunk 14]
```diff
@@ -15,152 +15,133 @@
...
+							</td>
+						<?php endif; ?>
+						<?php if ($this->params->get('show_state_headings')) : ?>
+							<td class="item-state" itemprop="address" itemscope itemtype="https://schema.org/PostalAddress">
+								<span itemprop="addressRegion"><?php echo $item->state; ?></span>
+							</td>
+						<?php endif; ?>
+						<?php if ($this->params->get('show_country_headings')) : ?>
+							<td class="item-state" itemprop="address" itemscope itemtype="https://schema.org/PostalAddress">
+								<span itemprop="addressCountry"><?php echo $item->country; ?></span>
```
Comment (Quy) [inline components/com_contact/views/featured/tmpl/default_items.php]: Split into multiple lines? If yes, do the same in the above lines?

**Comment by zero-24** [comment]
@franz-wohlkoenig please explain what you want from me? I'm a bit lost with your comment? :(

File: components/com_contact/views/contact/tmpl/default.php [hunk 15]
```diff
@@ -11,31 +11,34 @@
...
+				</span>
 			</h2>
 		</div>
 	<?php endif; ?>
-
-	<?php $show_contact_category = $tparams->get('show_contact_category'); ?>
-
 	<?php if ($show_contact_category === 'show_no_link') : ?>
 		<h3>
 			<span class="contact-category"><?php echo $this->contact->category_title; ?></span>
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default.php]: Multiple lines. Same for lines 49-50.

File: components/com_contact/views/contact/tmpl/default.php [hunk 16]
```diff
@@ -96,146 +98,114 @@
 				</dd>
 			</dl>
 		<?php endif; ?>
-
 		<?php echo $this->loadTemplate('address'); ?>
-
 		<?php if ($tparams->get('allow_vcard')) : ?>
 			<?php echo JText::_('COM_CONTACT_DOWNLOAD_INFORMATION_AS'); ?>
 			<a href="<?php echo JRoute::_('index.php?option=com_contact&amp;view=contact&amp;id=' . $this->contact->id . '&amp;format=vcf'); ?>">
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default.php:108]: Multiple lines.

File: components/com_contact/views/contact/tmpl/default_profile.php [hunk 17]
```diff
@@ -8,37 +8,34 @@
  */
 
 defined('_JEXEC') or die;
+
 ?>
 <?php if (JPluginHelper::isEnabled('user', 'profile')) :
 	$fields = $this->item->profile->getFieldset('profile'); ?>
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default_profile.php]: Coding style

File: components/com_contact/views/contact/tmpl/default_profile.php [hunk 18]
```diff
@@ -8,37 +8,34 @@
...
-							break;
-					endswitch;
-				endif;
-			endforeach; ?>
+			<?php foreach ($fields as $profile) : ?>
+				<?php if ($profile->value) : ?>
+					<?php echo '<dt>' . $profile->label . '</dt>'; ?>
+					<?php $profile->text = htmlspecialchars($profile->value, ENT_COMPAT, 'UTF-8'); ?>
+					<?php switch ($profile->id) : 
+						 case 'profile_website': ?>
```
Comment (Quy) [inline components/com_contact/views/contact/tmpl/default_profile.php:22]: Coding style?

File: components/com_contact/views/contact/tmpl/default.php (hunk 16 above)
Comment (zero-24) [reply components/com_contact/views/contact/tmpl/default.php:108]: why? This line looks good to me?

File: components/com_contact/views/contact/tmpl/default_profile.php (hunk 18 above)
Comment (zero-24) [reply components/com_contact/views/contact/tmpl/default_profile.php:22]: the first switch case can not be written different sadly :(

File: components/com_contact/views/contact/tmpl/default.php (hunk 16 above)
Comment (Quy) [reply components/com_contact/views/contact/tmpl/default.php:108]: line 105: Indent and move `</a>` to the next line.

Comment (zero-24) [reply components/com_contact/views/contact/tmpl/default.php:108]: fixed

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 2efc0806dc9b21528ede8805a84e98f4f47e9644<br /><br />Code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17217">issues.joomla.org/tracker/joomla-cms/17217</a>.</sub>

**Comment by zero-24** [comment]
Same here I'm closing this as it is conflicting and this is going to be moved against 4.0-dev in a later PR. Thanks @Quy , @franz-wohlkoenig , @Heggi93 & @brianteeman your comments they are not going to be missing. I'm going to implement as much as possible directly in the new PR. Thanks for all the work on the reviews 👍 ❤️

**Pull request closed without merging**